File: task2.js
var d = parseInt(process.argv[2]);
var m_str = (process.argv[3]).toString();
var y = parseInt(process.argv[4]);

var feb = 28;

/*
if (y >= 1600) {
    if (y % 400 == 0) {
        feb = 29;
    } else if (y % 100 == 0) {
        feb = 28;
    } else if (y % 4 == 0) {
        feb = 29;
    } else {
        feb = 28;
    }
} else {
    if (y % 4 == 0) {
        feb = 29;
    } else {
        feb = 28;
    }
}
*/

if (y % 4 == 0) {
    feb = 29;
} 


var dinm1 = {1: 31,
            2: feb,
            3: 31,
            4: 30,
            5: 31,
            6: 30,
            7: 31,
            8: 31,
            9: 30,
            10: 31,
            11: 30,
            12: 31};

var dinm = {"января": 31,
            "февраля": feb,
            "марта": 31,
            "апреля": 30,
            "мая": 31,
            "июня": 30,
            "июля": 31,
            "августа": 31,
            "сентября": 30,
            "октября": 31,
            "ноября": 30,
            "декабря": 31};

var res = dinm[m_str] - d; 
if (res>=0) {
    switch (m_str) {
        case "января"  : m = 1; break;
        case "февраля" : m = 2; break;
        case "марта"   : m = 3; break;
        case "апреля"  : m = 4; break;
        case "мая"     : m = 5; break;
        case "июня"    : m = 6; break;
        case "июля"    : m = 7; break;
        case "августа" : m = 8; break;
        case "сентября": m = 9; break;
        case "октября" : m = 10; break;
        case "ноября"  : m = 11; break;
        case "декабря" : m = 12; break;
        default        : m = 0;
    }

    var i=m;
    if (m < 12) {
        while (i<12) {
            res = res + dinm1[i];
            i++;
        }
    }

    console.log(res.toString());
}

File: task2.rb
d = ARGV[0].to_i
m_str = ARGV[1].to_s
y = ARGV[2].to_i
if (y >= 1600)
    if (y % 400 == 0)
        feb = 29
    elsif (y % 100 == 0) 
        feb = 28
    elsif (y % 4 == 0) 
        feb = 29
    else
        feb = 28
    end
else
    if (y % 4 == 0) 
        feb = 29
    else
        feb = 28
    end
end

dinm1 = Hash[1, 31,
            2, feb,
            3, 31,
            4, 30,
            5, 31,
            6, 30,
            7, 31,
            8, 31,
            9, 30,
            10, 31,
            11, 30,
            12, 31]

dinm = Hash["января", 31,
            "февраля", feb,
            "марта", 31,
            "апреля", 30,
            "мая", 31,
            "июня", 30,
            "июля", 31,
            "августа", 31,
            "сентября", 30,
            "октября", 31,
            "ноября", 30,
            "декабря", 31]
#puts dinm[m_str].to_s

res = dinm[m_str] - d 
if res>=0 
    case m_str
    when "января"
        m = 1
    when "февраля"
        m = 2
    when "марта"
        m = 3
    when "апреля"
        m = 4
    when "мая"
        m = 5
    when "июня"
        m = 6
    when "июля"
        m = 7
    when "августа"
        m = 8
    when "сентября"
        m = 9
    when "октября"
        m = 10
    when "ноября"
        m = 11
    when "декабря"
        m = 12
    else
      m = 0
    end
 
    i=m
    if m < 12
        loop do
            i+=1
            res = res + dinm1[i]
            break if i>=12
        end
    end

    puts res.to_s
end
